parser/filter.go

```diff
@@ -0,0 +1,164 @@
+package parser
+
+import (
+	"fmt"
+	"net"
+
+	"github.com/activecm/rita/resources"
+)
+
+func (fs *FSImporter) filterConnPair(src string, dst string) (ignore bool) {
+	// default is not to ignore connection pair
+	ignore = false
+
+	// parse src and dst IPs
+	srcIP := net.ParseIP(src)
+	dstIP := net.ParseIP(dst)
+
+	// check if on always included list
+	isSrcIncluded := fs.isAlwaysIncluded(srcIP)
+	isDstIncluded := fs.isAlwaysIncluded(dstIP)
+
+	// check if on never included list
+	isSrcExcluded := fs.isNeverIncluded(srcIP)
+	isDstExcluded := fs.isNeverIncluded(dstIP)
+
+	// if a result is on both lists we don't ignore it (need error handling later)
+	if (isSrcIncluded && isSrcExcluded) || (isDstIncluded && isDstExcluded) {
+		ignore = false
+		return
+	}
+
+	// check if one of the addresses should never be excluded from results
+	if isSrcExcluded || isDstExcluded {
+		ignore = true
+		return
+	}
+
+	// check if one of the addresses should never be excluded from results
+	if isSrcIncluded || isDstIncluded {
+		ignore = false
+		return
+	}
+
+	// verify if src and dst are internal or external
+	isSrcInternal := fs.isInternalAddress(srcIP)
+	isDstInternal := fs.isInternalAddress(dstIP)
+
+	// if both addresses are internal, filter applies
+	if isSrcInternal && isDstInternal {
+		ignore = true
+		return
+	}
+
+	// if both addresses are external, filter applies
+	if (!isSrcInternal) && (!isDstInternal) {
+		ignore = true
+		return
+	}
+
+	return
+}
+
+// Get internal subnets from the config file
+// todo: Error if a valid CIDR is not provided
+func getInternalSubnets(res *resources.Resources) []*net.IPNet {
+	var internalIPSubnets []*net.IPNet
+
+	internalFilters := res.Config.S.Filtering.InternalSubnets
+	for _, cidr := range internalFilters {
+		_, block, err := net.ParseCIDR(cidr)
+		internalIPSubnets = append(internalIPSubnets, block)
+		if err != nil {
+			fmt.Println("Error parsing config file CIDR.")
+			fmt.Println(err)
+		}
+	}
+	return internalIPSubnets
+}
+
+// Get "always included" subnets from the config file
+func getIncludedSubnets(res *resources.Resources) (includedSubnets []*net.IPNet) {
```

Comment:
Since the code is largely the same for all three `get*Subnets` functions, could we just combine them to something like this? I realize the internal function doesn't do the add '/32' on the end check but I don't see why it couldn't?
```go
// (I'm not sure whether *[]string or []*string is right)
// (and maybe we should add a logger too?)
 func getSubnets(subnetConfig *[]string) (subnets []*net.IPNet) {
  	for _, entry := range subnetConfig {
 		//try to parse out cidr range
 		_, block, err := net.ParseCIDR(entry)
 
  		//if there was an error, check if entry was an IP not a range
 		if err != nil {
 			// try to parse out IP as range of single host
 			_, block, err = net.ParseCIDR(entry + "/32")
 
  			// if error, report and return
 			if err != nil {
 				fmt.Println("Error parsing config file entry.")
 				fmt.Println(err)
 				return
 			}
 		}
 
  		// add cidr range to list
 		subnets = append(subnets, block)
 	}
 	return
 }
```

And call it like:
```go
internalSubnets := getSubnets(res.Config.S.Filtering.InternalSubnets)
neverSubnets := getSubnets(res.Config.S.Filtering.NeverInclude)
alwaysSubnets := getSubnets(res.Config.S.Filtering.AlwaysInclude)
```

Reply:
done!